internal/tiltfile/os/os.go

```diff
@@ -83,18 +104,60 @@ func cwd(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs [
 	return starlark.String(dir), nil
 }
 
-func realpath(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	var path string
-	err := starkit.UnpackArgs(t, fn.Name(), args, kwargs,
-		"path", &path,
-	)
-	if err != nil {
+// Add a function that takes exactly one parameter, a path string.
+func addPathBuiltin(env *starkit.Environment, name string,
+	f func(t *starlark.Thread, s string) (starlark.Value, error)) error {
+	builtin := func(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+		var path string
+		err := starkit.UnpackArgs(t, fn.Name(), args, kwargs,
+			"path", &path,
+		)
+		if err != nil {
+			return nil, err
+		}
+		return f(t, path)
+	}
+	return env.AddBuiltin(name, starkit.Function(builtin))
+}
+
+func abspath(t *starlark.Thread, path string) (starlark.Value, error) {
+	return starlark.String(starkit.AbsPath(t, path)), nil
+}
+
+func basename(t *starlark.Thread, path string) (starlark.Value, error) {
+	return starlark.String(filepath.Base(path)), nil
+}
+
+func dirname(t *starlark.Thread, path string) (starlark.Value, error) {
+	return starlark.String(filepath.Dir(path)), nil
+}
+
+func exists(t *starlark.Thread, path string) (starlark.Value, error) {
+	_, err := os.Stat(starkit.AbsPath(t, path))
```

Comment:
should this watch the path?

Reply:
I'm...honestly not sure. I could go either way on this. what do you think?

Reply:
I think it should. Imagine this:

```python
if os.exists('path/to/repo'):
  include('path/to/repo/Tiltfile')
```

I think that sort of use case could be important for micorepo setups

Reply:
I think in general we want the Tiltfile to reexecute when any of its filedeps change.

I guess it'd be annoying if someone is doing something like:
```
if not os.exists('foo.txt'):
  # create foo.txt
```
since it'd execute twice. I don't have a good feeling for the use cases for this or whether this would be a common thing. Seems fine as-is and if someone wants the other behavior we can tell them to `watch_file`.

Reply:
Asking people to manually watch_file sgtm too

Reply:
ok, i made it watch by default, and we can see what happens!